### DISCUSSION TITLE
debug_node_input_outputs_utils

### DISCUSSION THREAD

**Discussion by david-macleod** [microsoft/onnxruntime#8463] [Other Q&A] [answered]
As per [this file](https://github.com/microsoft/onnxruntime/blob/8d737f977056444a307f1b7f0bcd402fba62d790/onnxruntime/core/framework/debug_node_inputs_outputs_utils.h) we can build onnx to allow debugging `--cmake_extra_defines onnxruntime_DEBUG_NODE_INPUTS_OUTPUTS=1` 

My questions are:
1. Why is this not enabled by default, is there a performance oveheard by using this?
2. Why does `ORT_DEBUG_NODE_IO_DUMP_SHAPE_DATA` default to 1 when all other default to 0? This is slightly counterintuitive and means we have to this env variable explicitly to 0 most of the time.

Thanks

**Comment by edgchen1** [comment, marked as answer]
1. Yes, and this overhead is incurred around running each Op.
2. Some historical context here: https://github.com/microsoft/onnxruntime/pull/5768. It was the original behavior before that environment variable was added. Feel free to submit an issue or PR if you want to change it. This is a debugging facility though - I'd expect it to be used if one actually wanted to see additional debug output. If most of the time you don't, consider building without this enabled to avoid the performance overhead.

**Reply by david-macleod** [reply]
Thanks!
